packages/provider/src/transports/base-provider-transport.ts

```diff
@@ -147,9 +156,13 @@ export abstract class BaseProviderTransport implements ProviderTransport {
       return
     }
 
+    if (message.type === ProviderMessageType.AUTHORIZED) {
```

Comment:
we should remove "AUTHORIZED" event, and just use same event/signal as CONNECT for this data. Otherwise it creates multiple codepaths and async issues. I've experienced a bug when trying to connect + auth from cold start and it never asks me to authorize -- or does it perhaps authorize for me automatically?